Support EntraId

JoshInnis opened the pull request:
Add the ability to use EntraId authentication  to login to SQL Server

Commit:
Support EntraId

Add the ability to use EntraId authentication  to login to SQL Server

pghosh2025 commented on vectordb_bench/backend/clients/mssql/config.py:
```diff
@@ -1,18 +1,80 @@
-
 MSSQL_CONNECTION_STRING_PLACEHOLDER="DRIVER={ODBC Driver 18 for SQL Server};SERVER=%s;DATABASE=%s;UID=%s;PWD=%s;LongAsMax=yes;Connect Timeout=30;TrustServerCertificate=Yes"
 
+MSSQL_ENTRA_CONNECTION_STRING_PLACEHOLDER="DRIVER={ODBC Driver 18 for SQL Server};SERVER=%s;DATABASE=%s;AUTHENTICATION=ActiveDirectoryMsi;UID=%s;LongAsMax=yes;Connect Timeout=30;Encrypt=yes;TrustServerCertificate=Yes"
```
Where is being used?

JoshInnis replied:
Remove

pghosh2025 commented on vectordb_bench/backend/clients/mssql/config.py:
```diff
@@ -1,18 +1,80 @@
         return {
-            "connection_string" : MSSQL_CONNECTION_STRING_PLACEHOLDER%(self.server, self.database, self.uid, pwd_str)
+            "connection_string": connection_string,
+            "attrs_before": {SQL_COPT_SS_ACCESS_TOKEN: token_struct}
```
Where is being used?

JoshInnis replied:
mssql.py line 37 and line 107

pghosh2025 commented on vectordb_bench/backend/clients/mssql/config.py:
```diff
@@ -1,18 +1,80 @@
+        
+        # 1. Get credentials and token using azure-identity
+        credential = ManagedIdentityCredential(client_id=self.entraid)
+        access_token = credential.get_token(SQL_SERVER_TOKEN_SCOPE)
```
Check for valid credential and also access_token.

JoshInnis replied:
Currently, the project will throw an error:

`2025-09-02 15:56:38,793 | WARNING: [1/1] case {'label': <CaseLabel.Performance: 2>, 'dataset': {'data': {'name': 'OpenAI', 'size': 50000, 'dim': 1536, 'metric_type': <MetricType.COSINE: 'COSINE'>}}, 'db': 'MSSQL'} failed to run, reason=ManagedIdentityCredential authentication unavailable, no response from the IMDS endpoint. invalid_request (interface.py:199) (1002322)`

if the credentials are invalid. no additional checks are needed to avoid a segmentation fault. Only if we wish to alter the login message

Commit:
Cleanup Connection Strings

Commit:
Change logging and error checking to logging library

The rest of the project uses the logging library to report info and
errors. Switch the the changes on this branch to use the logging
library for these purposes too.
